classes/DepositObjectDAO.inc.php

```diff
@@ -161,8 +162,9 @@ public function createNew($journalId, $objectType) {
 		$objects = array();
 
 		switch ($objectType) {
-			case PLN_PLUGIN_DEPOSIT_OBJECT_ARTICLE:
-				$submissionDao = DAORegistry::getDAO('SubmissionDAO');
+			case 'PublishedArticle': // Legacy (OJS pre-3.2)
+			case PLN_PLUGIN_DEPOSIT_OBJECT_SUBMISSION:
+				$submissionDao = DAORegistry::getDAO('SubmissionDAO'); // Constants
 				$result = $this->retrieve(
 					'SELECT p.submission_id FROM publications p
```

Comment:
@asmecher I am not sure that this code is going to work for versions prior to 3.2, because of the use of `publications` table.

Reply:
No, you're correct, it won't work for anything prior to 3.2. There are specific commits for the 3.1.x to 3.2.x compatibility changes, and they would need to be reverted to create a `stable-3_1_2` branch.